### PR TITLE
Polkapool v2 

### PR DESCRIPTION
**Pull request by minikas**
## Description

This pull request introduces Polkapool v2 by integrating tools developed by the Polkadex team. The goal is to elevate the user experience and streamline functionality while acknowledging the foundational work done by CrusterCuss in the previous version.

## Changes
- [x] Integration of Polkadex Tools: Incorporates @polkadex/ux and @polkadex/ts developed by the Polkadex team to enhance usability and streamline functionality.
- [x] Refined User Interface: Improves the user interface to provide a smoother and more intuitive experience for users navigating the platform.
- [x] Optimized Performance: Implements optimizations to enhance performance and responsiveness, ensuring a seamless user experience even during peak usage.

## Screenshot

https://github.com/Polkadex-Substrate/polkadex-ts/assets/12574469/fbe3537d-732b-4e3b-8d2a-eb735b823f23

Close: https://github.com/Polkadex-Substrate/polkadex-ts/issues/203

## Summary by CodeRabbit

- **New Features**
    - Added liquidity addition functionality in the decentralized exchange application.
    - Added a custom React hook for fetching and calculating asset prices for swaps.
    - Implemented a swapping feature allowing asset exchanges with specified slippage.

- **Enhancements**
    - Configured Next.js settings for improved project setup.
    - Updated package dependencies and scripts for better performance and compatibility.

- **Refactor**
    - Defined custom styling configurations using Tailwind CSS for enhanced visual appearance.

- **Documentation**
    - Updated documentation for new functionalities and components.

- **Bug Fixes**
    - Ensured components handle loading states and user inputs effectively.

### CODE REVIEW

File: apps/polkapool/src/components/selectReceiveAsset.tsx [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+    () =>
+      poolsReserve
+        ?.filter((c) => {
+          if (c.id === "POLKADEX") return c;
+          return pools?.some((d) => c.id === d.quote.id);
+        })
+        ?.sort(
+          (a, b) => (a.id === baseId ? -1 : 0) - (b.id === baseId ? -1 : 0)
+        ),
+    [poolsReserve, baseId, pools]
```
[the remark this answers is not in the document]
Comment (minikas) [reply apps/polkapool/src/components/selectReceiveAsset.tsx:45]: Unnecessary dependency

File: apps/polkapool/src/components/selectPayAsset.tsx [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+      assets
+        ?.filter((c) => {
+          if (c.id === "POLKADEX") return c;
+          return pools?.some((d) => c.id === d.quote.id);
+        })
+        ?.sort(
+          (a, b) => (a.id === baseId ? -1 : 0) - (b.id === baseId ? -1 : 0)
+        ),
+    [assets, baseId, pools]
+  );
```
[the remark this answers is not in the document]
Comment (minikas) [reply apps/polkapool/src/components/selectPayAsset.tsx:37]: Unnecessary dependency

File: apps/polkapool/src/constants/parseScientific.ts
Comment (ap211unitech) [inline apps/polkapool/src/constants/parseScientific.ts:1]: Please add this function in `@polkadex/numericals` and import from there

File: apps/polkapool/src/constants/trimFloat.ts
Comment (ap211unitech) [inline apps/polkapool/src/constants/trimFloat.ts:1]: This function is already present in `@polkadex/numericals`. Please import it from there.

File: apps/polkapool/src/helpers/truncateString.ts
Comment (ap211unitech) [inline apps/polkapool/src/helpers/truncateString.ts:1]: Same here.

**Review by serhii-temchenko — APPROVED** [review]

**Review by ap211unitech — APPROVED** [review]

**Pull request merged**